pkg/ovndbcluster/dnsdata.go

```diff
@@ -0,0 +1,78 @@
+package ovndbcluster
+
+import (
+	"context"
+	"fmt"
+
+	infranetworkv1 "github.com/openstack-k8s-operators/infra-operator/apis/network/v1beta1"
+	"github.com/openstack-k8s-operators/lib-common/modules/common/helper"
+	"github.com/openstack-k8s-operators/ovn-operator/api/v1beta1"
+	ovnv1 "github.com/openstack-k8s-operators/ovn-operator/api/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+// Create DNSRecord
+func DnsData(
```

Comment:
as mentioned elsewhere, if your function has action (create or patch), then the name should contain a verb (EnsureDnsData?). Or - better? - separate the code to produce the object definition from CreateOrPatch call (see how we have StatefulSet function to produce a SS object definition but not create it.)